### PR TITLE
window: cleanup delete-event handling

### PR DESCRIPTION
**Pull request by daschuer**
Always handle it in a class handler.

Nautilus commit c00d47b32c60869d6dcdfd1e15368b3a22ac3fcb

### CODE REVIEW

File: src/nemo-desktop-window.h [hunk 1]
```diff
@@ -56,7 +55,7 @@ typedef struct {
 
 GType                  nemo_desktop_window_get_type            (void);
 NemoDesktopWindow     *nemo_desktop_window_new                 (gint monitor);
-void                   nemo_desktop_window_update_directory    (NemoDesktopWindow *window);
+void                   nemo_desktop_window_ensure              (void);
```
Comment (itzexor) [inline src/nemo-desktop-window.h]: What is this?

File: src/nemo-desktop-window.c [hunk 2]
```diff
@@ -209,17 +202,15 @@ nemo_desktop_window_new (gint monitor)
...
 
-    g_signal_connect (window, "delete_event", G_CALLBACK (nemo_desktop_window_delete_event), NULL);
-
-    GdkRGBA transparent = {0, 0, 0, 0};
-    gtk_widget_override_background_color (GTK_WIDGET (window), 0, &transparent);
-
-	/* Point window at the desktop folder.
-	 * Note that nemo_desktop_window_init is too early to do this.
-	 */
-	nemo_desktop_window_update_directory (window);
```
Comment (itzexor) [inline src/nemo-desktop-window.c]: Why is this whole section being removed?

File: src/nemo-desktop-window.h [hunk 3]
```diff
@@ -47,7 +47,6 @@ typedef struct NemoDesktopWindowDetails NemoDesktopWindowDetails;
 typedef struct {
 	NemoWindow parent_spot;
 	NemoDesktopWindowDetails *details;
-        gboolean affect_desktop_on_next_location_change;
```
Comment (itzexor) [inline src/nemo-desktop-window.h]: This also seems unrelated.

**Comment by daschuer** [comment]
Ups ...  something went wrong. I will try again.

**Comment by daschuer** [comment]
OK, I have fixed the issues and force pushed this branch.

File: src/nemo-desktop-window.h (hunk 3 above)
Comment (daschuer) [inline src/nemo-desktop-window.h]: affect_desktop_on_next_location_change is not used. So we can keep this unrelated change.
